### PR TITLE
markers plugin: added support for custom marker elements

### PR DESCRIPTION
**Pull request by bknower**
### Short description of changes:
Markers can now be provided with an image to display instead of the default pointer image when being added.

### CODE REVIEW

**Comment by thijstriemstra** [comment]
good idea. what if someone wants to add videos, we'd have to add another parameter to that method, I'd prefer a different approach that would make it easy to support various types (eg audio/video/img).

**Comment by bknower** [comment]
> good idea. what if someone wants to add videos, we'd have to add another parameter to that method, I'd prefer a different approach that would make it easy to support various types (eg audio/video/img).

Does it make sense to display anything other than an image as a marker? This is just for the markers like this 
image

**Review by thijstriemstra — CHANGES_REQUESTED** [review]
> Does it make sense to display anything other than an image as a marker?

Yeah, a canvas element. But also an html element like a button or checkbox. Can you make it more generic? Also, can you add this new feature to the markers example?

**Review by thijstriemstra — CHANGES_REQUESTED** [review]
thanks. can you also document the new option here: https://github.com/katspaugh/wavesurfer.js/blob/554ba967c8da3e271fab911c4ca8b0eaa4260806/src/plugin/markers/index.js#L3

And add a changelog entry?

**Comment by osheroff** [comment]
needs documentation.  Interface is highly flexible but seems ripe for shooting yourself in the foot by re-using the same HTMLElement; I could see the argument for detecting that case?  I was going to say "just use `cloneNode`" but maybe that hampers some use cases.   Dunno.

@thijstriemstra is there an example of a plugin accepting an HTMLElement we could follow?

**Comment by thijstriemstra** [comment]
> is there an example of a plugin accepting an HTMLElement we could follow?

Not that I know of. I think this PR is good enough for a first implementation.

File: CHANGES.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ wavesurfer.js changelog
 ------------------
 
 - Fix removing DOM element on `destroy()` (#2258)
+- Markers plugin: add the ability to use custom HTML elements in place of the default marker icon by passing the new `markerElement` parameter to the marker constructor (#2269)
```
Comment (thijstriemstra) [inline CHANGES.md:8]: move this changelog entry to a new section called `x.x.x (unreleased)`

**Review by thijstriemstra — APPROVED** [review]
thanks.

**Pull request merged**